packages/@aws-cdk/aws-codebuild/lib/project.ts

```diff
@@ -764,7 +774,25 @@ export class Project extends ProjectBase {
 
         // save SecretsManager env variables
         if (envVariable.type === BuildEnvironmentVariableType.SECRETS_MANAGER) {
-          if (Token.isUnresolved(envVariableValue)) {
+          if (SecretValue.isSecretValue(envVariable.value)) {
```

Comment:
I don't think we need any of this.

Let's just make the `envVariableValue` be the string that contains the `secretId` from `SecretValue` above (concatenated with the qualifiers like `jsonField`, `versionStage`, `versionId`). Then, the existing logic that handles this below will correctly parse it, and we will be able to remove this entire `if` (which is duplicated with the non-Token case).